packages/actor-rdf-resolve-quad-pattern-string-source/test/ActorRdfResolveQuadPatternStringSource-test.ts

```diff
@@ -0,0 +1,203 @@
+      return expect(actor.test({ pattern: <any> null,
+        context: new ActionContext(
+          { [KeysRdfResolveQuadPattern.source.name]:
+            { type: 'stringSource', value: rdfSource, mediaType: sourceMediaType }},
+        ) })).rejects.toBeTruthy();
+    });
+  });
+
+  describe('The run method of an ActorRdfResolveQuadPatternRdfJsSource instance', () => {
+    let actor: ActorRdfResolveQuadPatternStringSource;
+    const expectedQuads: RDF.Quad[] = [
+      DF.quad(DF.namedNode('ex:a'), DF.namedNode('ex:b'), DF.literal('c')),
+      DF.quad(DF.namedNode('ex:a'), DF.namedNode('ex:d'), DF.literal('e')),
+    ];
+
+    beforeEach(() => {
+      mockMediatorRdfParse = {
+        mediate(_arg: any) {
+          return {
+            handle: {
+              // StreamifyArray has the side effect of comsuming the object hence the clone operation
+              data: streamifyArray(JSON.parse(JSON.stringify(expectedQuads))),
+            },
+          };
+        },
+      };
+      mockMediatorRdfQuadPattern = {
+        mediate(_arg: any) {
+          return {
+            // StreamifyArray has the side effect of comsuming the object hence the clone operation
+            data: streamifyArray(JSON.parse(JSON.stringify(expectedQuads))),
+          };
+        },
+      };
+      spyMockMediatorRdfParse = jest.spyOn(mockMediatorRdfParse, 'mediate');
+      spyMockMediatorRdfQuadPattern = jest.spyOn(mockMediatorRdfQuadPattern, 'mediate');
+      actor = new ActorRdfResolveQuadPatternStringSource({ name: 'actor',
+        bus,
+        mediatorRdfParse: mockMediatorRdfParse,
+        mediatorRdfQuadPattern: mockMediatorRdfQuadPattern });
+    });
+
+    it('should run', async() => {
+      const context = new ActionContext({ name: 'context',
+        [KeysRdfResolveQuadPattern.source.name]:
+      { type: 'stringSource', value: sourceValue, mediaType: sourceMediaType, baseIri: sourceBaseIri }});
+      const pattern: any = {
+        subject: DF.variable('s'),
+        predicate: DF.namedNode('p'),
+        object: DF.variable('o'),
+        graph: DF.variable('g'),
+      }; // It can be anything the mediator returned always the same triples
+      const op: any = {
+        context,
+        pattern: {
+          context,
+          pattern,
+        },
+      };
+
+      const expectedTextStream = streamifyString(<string> sourceValue);
+      const expectedParseAction = {
+        context,
+        handle: {
+          metadata: { baseIRI: sourceBaseIri },
+          data: expectedTextStream,
+          context,
+        },
+        handleMediaType: sourceMediaType,
+      };
+
+      const resp: IActorRdfResolveQuadPatternOutput = await actor.run(op);
+      expect(spyMockMediatorRdfParse).toBeCalledWith(expect.objectContaining(expectedParseAction));
+
+      expect(spyMockMediatorRdfQuadPattern).toBeCalledWith(expect.objectContaining({
+        pattern: op.pattern,
+      }));
+      expect(resp).toBeDefined();
```

Comment:
strangely the CI test had a hard time to recognize asynciterator, maybe because of the interference between the native one

Reply:
What was the problem exactly?
Unlikely that there would be interference between the native asynciterator.

Reply:
What part of the CI says that? The unit tests or the linter?

Reply:
Unit test ; CI / test (ubuntu-latest, 14.x) (pull_request)